### PR TITLE
Url assembly with query

### PR DESCRIPTION
**Pull request by stackoverfloweth**
this ones pretty straightforward, finally

### CODE REVIEW

File: src/utilities/urlAssembly.spec.ts [hunk 1]
```diff
@@ -1,83 +1,170 @@
...
-  ['/simple/:simple'],
-  [path('/simple/:simple', { simple: String })],
-])('given route with required string param provided, returns route Path with string with values interpolated', (path) => {
-  const route = {
-    name: 'simple',
-    path,
-    component,
-  } satisfies Route
-  const [resolved] = resolveRoutes([route])
+describe('query params', () => {
```
Comment (stackoverfloweth) [inline src/utilities/urlAssembly.spec.ts:87]: here are the new tests

File: src/utilities/urlAssembly.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import { Param, Resolved, Route } from '@/types'
 import { setParamValuesOnUrl } from '@/utilities/paramsFinder'
 
-export function assembleUrl(route: Resolved<Route>, values: Record<string, unknown> = {}): string {
+export function assembleUrl(route: Resolved<Route>, values: Record<string, unknown[]> = {}): string {
   const params = Object.entries<Param[]>(route.params)
+  const pathWithQuery = route.query.length ? `${route.path}?${route.query}` : route.path
```
Comment (stackoverfloweth) [inline src/utilities/urlAssembly.ts:6]: I considered moving this out into it's own utility

File: src/utilities/urlAssembly.spec.ts [hunk 3]
```diff
@@ -1,83 +1,170 @@
...
+      name: 'simple',
+      path: '/',
+      query,
+      component,
+    } satisfies Route
+    const [resolved] = resolveRoutes([route])
+
+    const url = assembleUrl(resolved, {})
+
+    expect(url).toBe('/?simple=')
```
Comment (pleek91) [inline src/utilities/urlAssembly.spec.ts:119]: I'm not sure what the expected behavior here would be. This test makes sense for sure. But I wonder if users will expect the key to not be present when the value is absent. But I honestly like how simple this is.

File: src/utilities/urlAssembly.ts (hunk 2 above)
Comment (pleek91) [reply src/utilities/urlAssembly.ts:6]: This whole utility is pretty simple. So this ternary doesn't bother me.

File: src/utilities/urlAssembly.ts [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
 
   return params.reduce((url, [name, params]) => {
     return setParamValuesOnUrl(url, { name, params, values: values[name] })
```
Comment (pleek91) [inline src/utilities/urlAssembly.ts:9]: Super cool you didn't actually have to write any additional logic for filling in the params

**Review by pleek91 — APPROVED** [review]

File: src/utilities/urlAssembly.ts (hunk 4 above)
Comment (stackoverfloweth) [reply src/utilities/urlAssembly.ts:9]: yeah, kudos to you for finding the syntax that just works and feels good

File: src/utilities/urlAssembly.spec.ts (hunk 3 above)
Comment (stackoverfloweth) [reply src/utilities/urlAssembly.spec.ts:119]: agreed. And unless you felt strongly I was going to just go with whatever is the most cut and dry and let our users tell us what improvements they want to see here

**Pull request merged**